packages/react-query/src/bin/postinstall.ts

```diff
@@ -0,0 +1,22 @@
+import { getTanStackReactQueryPackageJson } from './utils/package'
+import { switchVersion } from './utils/switchVersion'
+
+try {
+  const { version } = getTanStackReactQueryPackageJson()
+
+  if (!version) {
+    throw new Error()
+  }
+
+  if (version.startsWith('4.')) {
+    switchVersion(4)
+  } else if (version.startsWith('5.')) {
+    switchVersion(5)
+  } else {
+    console.warn('[@suspensive/react-query]', `version v${version} is not supported.`)
+  }
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+} catch (error) {
+  console.error('[@suspensive/react-query]', 'not fond @tanstack/react-query')
+  process.exit(1)
+}
```

Comment:
Original code
```ts
import { getTanStackReactQueryPackageJson } from './utils/package'
import { switchVersion } from './utils/switchVersion'

const { version } = getTanStackReactQueryPackageJson()

if (version.startsWith('4.')) {
  switchVersion(4)
} else if (version.startsWith('5.')) {
  switchVersion(5)
} else {
  console.warn('[@suspensive/react-query]', `version v${version} is not supported.`)
}
```

I suggest below code.

```suggestion
const { version } = getTanStackReactQueryPackageJson()

if (!version) {
  throw new Error('[@suspensive/react-query]', 'not found version of @tanstack/react-query')
}

if (version.startsWith('4.')) {
  switchVersion(4)
} else if (version.startsWith('5.')) {
  switchVersion(5)
} else {
  console.warn('[@suspensive/react-query]', `version v${version} is not supported.`)
}
```

if you want use try catch, you could make specific error constructor to specify that caught error is for not found version

```suggestion
class NotFoundVersionError extends Error {
  message: string = 'Not found version'
}

try {
  const { version } = getTanStackReactQueryPackageJson()

  if (!version) {
    throw new NotFoundVersionError()
  }

  if (version.startsWith('4.')) {
    switchVersion(4)
  } else if (version.startsWith('5.')) {
    switchVersion(5)
  } else {
    console.warn('[@suspensive/react-query]', `version v${version} is not supported.`)
  }
  // eslint-disable-next-line @typescript-eslint/no-unused-vars
} catch (error) {
  // but in my opinion, this is not good idea
  if(error instanceof NotFoundVersionError){ // specify error
    console.error('[@suspensive/react-query]', error.message)
  }
  process.exit(1)
}
```

Reply:
My intention was to simplify the postinstall script. Ultimately, I designed it so that if an error occurs at runtime (e.g., `node ./dist/bin/postinstall.cjs || echo 'please reinstall @suspensive/react-query'`), the error would propagate as intended. The goal was to consistently inform the user of an appropriate solution through a clear message.

Therefore, I believe this file does not cause any inconvenience to the user due to node(runtime) errors, and I will revert it to its previous state (this was my mistake).